### ISSUE TITLE
Add option to verify checksum without requiring write access to remote objects

### ISSUE THREAD

**Issue by ghuls** [kuleuven/mango-ingest#11]
Add option to verify checksum without requiring write access to remote objects and without requiring to upload new data.

I wanted to check some data uploaded by another user, which I can read, but with the current mango-ingest, I can't verify the checksums as it requires write permissions to remote object by default.

Trying to run the `--dry-run` option with `--verify-checksum` does not work for just checking the checksum as it will just print `dry-run: would upload /path/to/file`
```
$ mango_ingest.py -r -p local_dir/ -d /irods --verify-checksum --verbose --no-watch --dry-run
```

**Comment by paulborgermans** [comment]
Indeed, checksum operations require write access to the object by the user executing the code. That is not something that I/we can change as it is the way the irods backend works. Maybe a stricter checking of access rights should be added?

**Comment by ghuls** [comment]
I don't see why it shouldn't be possible to retrieve a precalculated checksum from iRODS as a user with read only rights. If that is really not possible it is something that should be fixed in the iRODS backend.
